code/game/machinery/computer/mechlaunchpad.dm

```diff
@@ -0,0 +1,155 @@
+/obj/machinery/computer/mechpad
+	name = "orbital mech pad console"
+	desc = "A computer designed to handle the calculations and routing required for sending and receiving mechs from orbit. Requires a link to a nearby Orbital Mech Pad to function."
+	icon_screen = "mechpad"
+	icon_keyboard = "teleport_key"
+	circuit = /obj/item/circuitboard/computer/mechpad
+	///Selected mechpad in the console
+	var/selected_id
+	///Mechpads that it can send mechs through to other mechpads
+	var/obj/machinery/mechpad/connected_mechpad
+	///List of mechpads connected
+	var/list/obj/machinery/mechpad/mechpads = list()
+	///Maximum amount of pads connected at once
+	var/maximum_pads = 3
+
+/obj/machinery/computer/mechpad/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		connected_mechpad = connect_to_pad()
+		connected_mechpad.connected_console = src
+		return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/computer/mechpad/LateInitialize()
+	for(var/obj/machinery/mechpad/pad in GLOB.machines)
+		if(pad == connected_mechpad)
+			continue
+		mechpads += pad
+		pad.consoles += src
+		if(mechpads.len > maximum_pads)
+			break
```

Comment:
This will connect the mechpad to random pads across the machines.
Give it some granularity. Instead, make a `GLOB.mechpads` associative list, and add a matching string id var to the mechpad machines and computers.
On mechpad machinery's `Initialize` you can do a `GLOB.mechpads[id] += list(src)`
On mechpad computer `LateInitialize` you can do something like this:
```cpp
for(var/p in GLOB.mechpads[id])
	var/obj/machinery/mechpad/pad
	...
```